apiserver/cmd/apiserver/server/tier_upgrade.go

```diff
@@ -0,0 +1,103 @@
+package server
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	v1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/klog/v2"
+
+	client "github.com/projectcalico/calico/libcalico-go/lib/clientv3"
+	"github.com/projectcalico/calico/libcalico-go/lib/options"
+	"github.com/projectcalico/calico/libcalico-go/lib/winutils"
+)
+
+func migratePolicyNames() error {
+	k8sconfig, err := winutils.BuildConfigFromFlags("", os.Getenv("KUBECONFIG"))
+	if err != nil {
+		return fmt.Errorf("failed to build kubernetes client config: %s", err)
+	}
+
+	// Get Kubernetes clientset
+	k8sClientset, err := kubernetes.NewForConfig(k8sconfig)
+	if err != nil {
+		return fmt.Errorf("failed to build kubernetes client: %s", err)
+	}
+	if !updateNeeded(k8sClientset) {
+		return nil
+	}
+	klog.Infof("Migrating policy names")
+	calicoClient, err := client.NewFromEnv()
+	if err != nil {
+		return err
+	}
+
+	networkPolicies, err := calicoClient.NetworkPolicies().List(context.Background(), options.ListOptions{})
+	if err != nil {
+		return err
+	}
+	for _, policy := range networkPolicies.Items {
+		policy.Name = removeDefaultTierPrefix(policy.Name)
+		_, err = calicoClient.NetworkPolicies().Update(context.Background(), &policy, options.SetOptions{})
+		if err != nil {
+			return err
+		}
+	}
+
+	globalNetworkPolicies, err := calicoClient.GlobalNetworkPolicies().List(context.Background(), options.ListOptions{})
+	if err != nil {
+		return err
+	}
+	for _, policy := range globalNetworkPolicies.Items {
+		policy.Name = removeDefaultTierPrefix(policy.Name)
+		_, err = calicoClient.GlobalNetworkPolicies().Update(context.Background(), &policy, options.SetOptions{})
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func removeDefaultTierPrefix(name string) string {
+	if strings.HasPrefix(name, "default.") {
+		name = strings.TrimPrefix(name, "default.")
+	}
+	return name
+}
+
+func updateNeeded(k8sClientset *kubernetes.Clientset) bool {
+	tierMigration, err := k8sClientset.CoreV1().ConfigMaps("calico-apiserver").Get(context.Background(), "api-server", metav1.GetOptions{})
```

Comment:
Strictly speaking we may be running in a different namespace in some environments - I believe you can learn what namespace you are in based on a projected file mounted into the container at `/var/run/secrets/kubernetes.io/serviceaccount/namespace` though. Might already be loaded by the go-client? I can't remember.

Reply:
Worth verifying that we have the necessary RBAC in our manifests / operator to get, create, and update this configmap.